src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -35,12 +39,28 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnBrokerChanged(object sender, EventArgs e) {
+        private void OnRSessionInitialized(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
```

Comment:
CPS has a total mess of over engineered dataflow blocks, MEF imports and async events, and I'm not trying to mimic it. And yes, `async/await` itself is a heavily leaking abstraction. 

None of these though makes C# events a silver bullet to control program flow:
- There is no good way to control how many times event must be handled. Great example is WPF 3.5, where you could easily get millions of `PropertyChanged` without an idea if handler should take care of current notification or skip it because the same property will send another `PropertyChanged`. Message pipes with predefined enumeration order are much better in these cases, and dataflow blocks suck because of the way they are implemented, not because pattern itself is bad. 
- There is no good way to get notified by event only once. This is our case. It is good to have a set of events that notify listeners about the state of the object, but in the multi-threaded environment, you sign for notification or execute handler if object is in the required state already, so code usually looks like this:
```cs
if (_obj.IsReady) {
    handle(_obj);
} else {
    _obj.Ready += HandleIsReady;
    if (_obj.IsReady) {
        _obj.Ready -= HandleIsReady;
        handle(_obj);
    }
}

private void HandleIsReady(object sender, EventArgs args) {
    _obj.Ready -= HandleIsReady;
    handle(_obj);
}
```
while with tasks and async/await, you need just 
```cs
await _obj.IsReady;
handle(_obj)
```
- Normal events do not allow raiser to wait for handlers in other threads without blocking the thread, because you can't automatically combine several delegates with non-void return type. You'll have to manually iterate through `MulticastDelegate` list of delegates and raise them one by one, ending up with CPS async event, where people forget to use special extension methods and get very unpleasant bugs. They've been hunting these bugs for years. 
- You'll also have to manually iterate through delegates if you want to swallow specific types of exceptions in failed handler and allow other handlers to proceed.

And so on. Every tool has its own scope and set of pros and cons. I don't see why we need to use event when there is a method that does everything we need by itself. On the other hand, if for #2736 event is better, I'll use an event.